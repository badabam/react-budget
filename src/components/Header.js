import React, { Component } from 'react'
import styled from 'react-emotion'
import { Link } from 'react-router-dom'

const StyledHeader = styled('header')`
  grid-row: 1 / auto;
  display: flex;
  align-items: center;
  justify-content: center;
  background: cornflowerblue;
  color: white;
  font-size: 1.2em;
  position: relative;
`

const HeaderLink = styled(Link)`
  color: white;
  font-size: 0.8em;
  position: absolute;
`

export default class Header extends Component {
  static defaultProps = {
    link: { left: true },
  }

  render() {
    const { url: linkUrl, text: linkText, left: isLinkLeft } = this.props.link

    const linkPositionStyle = {
      right: isLinkLeft ? 'auto' : '10px',
      left: isLinkLeft ? '10px' : 'auto',
    }

    return (
      <StyledHeader>
        {linkUrl &&
          linkText && (
            <HeaderLink style={linkPositionStyle} to={linkUrl}>
              {linkText}
            </HeaderLink>
          )}
        {this.props.children}
      </StyledHeader>
    )
  }
}
