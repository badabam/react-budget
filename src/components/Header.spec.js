import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'
import { MemoryRouter } from 'react-router-dom'
import toJson from 'enzyme-to-json'

describe('Header', () => {
  let wrapper
  let link = { url: '/test', text: 'foo' }

  it('renders its children', () => {
    wrapper = shallow(
      <Header>
        <div>Hello</div>
      </Header>
    )
    const innerHtml = wrapper.html()
    expect(innerHtml).toContain('<div>Hello</div>')
  })

  it('renders its links', () => {
    wrapper = shallow(
      // to prevent errors when using Router in the app,
      // we need to wrap our components in MemoryRouter
      <MemoryRouter>
        <Header link={link}>
          <div>World</div>
        </Header>
      </MemoryRouter>
    )
    const innerHtml = wrapper.html()
    expect(innerHtml).toContain('href="/test"')
    expect(innerHtml).toContain('foo</a>')
    expect(innerHtml).toContain('right:10px;left:auto')
  })

  it('renders links also on the left', () => {
    wrapper = shallow(
      // to prevent errors when using Router in the app,
      // we need to wrap our components in MemoryRouter
      <MemoryRouter>
        <Header link={{ ...link, left: true }}>
          <div>World</div>
        </Header>
      </MemoryRouter>
    )
    const innerHtml = wrapper.html()
    expect(innerHtml).toContain('href="/test"')
    expect(innerHtml).toContain('foo</a>')
    expect(innerHtml).toContain('right:auto;left:10px')
  })

  it('keeps it snapshot', () => {
    wrapper = shallow(<Header link={link}>test</Header>)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ link: { ...link, left: true } })
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
