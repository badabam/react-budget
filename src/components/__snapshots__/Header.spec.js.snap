// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header keeps it snapshot 1`] = `
<Styled(header)>
  <Styled(Link)
    style={
      Object {
        "left": "auto",
        "right": "10px",
      }
    }
    to="/test"
  >
    foo
  </Styled(Link)>
  test
</Styled(header)>
`;

exports[`Header keeps it snapshot 2`] = `
<Styled(header)>
  <Styled(Link)
    style={
      Object {
        "left": "10px",
        "right": "auto",
      }
    }
    to="/test"
  >
    foo
  </Styled(Link)>
  test
</Styled(header)>
`;
